lib/screens/events/events_repository.dart

```diff
@@ -52,3 +57,36 @@ class FakeEventsRepository implements EventsRepository {
     }
   }
 }
+
+class APIEventsRepository implements EventsRepository {
+  final String classTag = "APIEventsRepository";
+  @override
+  Future<List<dynamic>> events() async {
```

Comment:
1. As there will be a list of events and not one event. So it will be better if we use eventModel at the UI screen just to extract data from json rather using eventModel here.

Reply:
2. Done

Reply:
as per the core principle, the reason of using cubit (/bloc), is that ui and business logic should be completely separated. Converting json to dart eventModel object is a business logic, to be done in cubit itself. It will reduce redundant data (values in the json map that are not useful for the app) flowing throw the app.

Reply:
Done